### ISSUE TITLE
History not loading when opening conversation from notification

### ISSUE THREAD

**Issue by mbirth** [iNPUTmice/Conversations#2238]
#### General information

* **Version:** 1.15.4
* **Device:** Blackberry PRIV
* **Android Version:** Android 6.0.1 Stock
* **Server name:** self hosted
* **Server software:** ejabberd 16.12-beta1 (https://hub.docker.com/r/ejabberd/ecs/)
* **Installed server modules:** various, e.g. carboncopy, last, mam, muc, offline, proxy65, etc.
* **Conversations source:** PlayStore Beta Channel

#### Steps to reproduce

Difficult to catch. It happens when I talked to someone via my desktop PC for a good while, then leave the desktop and get a message so Conversations notifies me. Also it seems to only happen to threads other than the one I've currently opened in Conversations.

#### Result

Conversations opens the thread with the person and shows only the just received message, the ones from before are missing (were never received, it seems). It doesn't try to populate the history with the missed messages.

**Comment by iNPUTmice** [comment]
Please always attach an adb logcat. Otherwise there is absolutely nothing I can do about this. Especially since this doesn't seem to be reproducible as you say.

**Comment by mbirth** [comment]
I'll see if I can reproduce it later today.

**Comment by mbirth** [comment]
I suspect it has to do with doze mode. Phone was lying on my desk for about 4 hours. Now when opening it (Conversations was still open), it was missing conversations (made from my desktop PC) from the last 4 hours. Now after picking it up (around 14:20), it didn't even try to load the history from the server. Only after disconnecting and reconnecting to the server (i.e. switching the connection off and on again in "Manage Accounts"), it pulled the history and was up to date again.

[logcat.txt](https://github.com/siacs/Conversations/files/841424/logcat_anon.txt)

**Comment by colegota** [comment]
I got a similar issue too. I noticed first some weeks ago, but not sure when it started.
If I have Gajim working in my desktop, messages from other user accounts don't came to my phone. Nor Conversations, nor Xabber.
Messages to MUC rooms arrives both to Gajim/Desktop and Conversations/Xabber at phone.

In Conversations, if I disable the account and activate it again, it recovers messages from server and it works for some time again. But later it start to miss messages again.
In Xabber after disable and enable the account, messages are not recovered from server.
